test/compilable/commontype.d

```diff
@@ -299,11 +299,11 @@ static assert(is( X!( shared(const(C))[4], shared(C)[4] ) == shared(const(C))[4]
 static assert(is( X!( shared(C)[4], shared(const(C))[4] ) == shared(const(C))[4] ));
 
 // base class conversion
-static assert(Error!( C[4], B[4] ));
-static assert(Error!( C[4], I[4] ));
+static assert(is( X!(C[4], B[4]) ));
+static assert(is( X!(C[4], I[4]) ));
 static assert(Error!( C[4], D[4] ));
-static assert(is( X!( C[4], const(B)[4] ) == const(B)[] )); // !?
-static assert(Error!( C[4], const(I)[4] ));
+static assert(is( X!( C[4], const(B)[4] ) == const(B)[4] ));
+static assert(is( X!( C[4], const(I)[4] )));
```

Comment:
I'm not sure how to tackle this. If we start with `testA` can you explain how the behaviour should change?

With this change I get

```
core.exception.AssertError@runnable/iconv_interface_array.d(18): 23 != 0
```

instead of yours: `13 != 0`.

Reply:
I've disabled conversion to interfaces for now. Can you take a look @MoonlightSentinel?

Reply:
That seems to be the best solution because otherwise you would need to emit code which adjusts the offset for each array element IIUC.

> instead of yours: 13 != 0.

Well, both values are garbage, so differences are to be expected.

> . If so, what should it look like?

I think the current `errorSupplemental` (`cannot pass argument arr of type C[4] to parameter I1[4] arr`) should be good enough. You could also add something like  `Cannot implicitly convert C[4] to I1[4]` but that is redundant IMHO.

Reply:
Ok, great. Let's see if things go green now and then I'll update the spec.